lib/membrane_ffmpeg_video_filter/text_overlay.ex

```diff
@@ -0,0 +1,151 @@
+              fontcolor: [
+                type: :binary,
+                description:
+                  "Choose font color according to the ffmpeg color syntax (https://ffmpeg.org/ffmpeg-utils.html#color-syntax)",
+                default: "black"
+              ],
+              fontfile: [
+                type: :binary,
+                description:
+                  "Path to the file with the desired font. If not set, default font fallback from fontconfig is used",
+                default: nil
+              ],
+              box?: [
+                type: :boolean,
+                description: "Set to true if a box is to be displayed behind the text",
+                default: false
+              ],
+              boxcolor: [
+                type: :binary,
+                description: "If the box? is set to true, display a box in the given color",
+                default: "white"
+              ],
+              border?: [
+                type: :boolean,
+                description: "Set to true to display a gray border around letters",
+                default: false
+              ],
+              vertical_align: [
+                type: :atom,
+                spec: :left | :right | :center,
+                description: "Horizontal position of the displayed text",
+                default: :left
+              ],
+              horizontal_align: [
+                type: :atom,
+                spec: :top | :bottom | :center,
+                description: "Vertical position of the displayed text",
+                default: :bottom
+              ]
+
+  def_input_pad :input,
+    demand_unit: :buffers,
+    caps: {Raw, aligned: true}
+
+  def_output_pad :output,
+    caps: {Raw, aligned: true}
+
+  @impl true
+  def handle_init(options) do
+    state =
+      options
+      |> Map.from_struct()
+      |> Map.put(:native_state, nil)
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_demand(:output, _size, :buffers, _ctx, %{native_state: nil} = state) do
+    # Wait until we have native state (after receiving caps)
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_demand(:output, size, :buffers, _ctx, state) do
+    {{:ok, demand: {:input, size}}, state}
+  end
+
+  @impl true
+  def handle_caps(
+        :input,
+        %Raw{format: format, width: width, height: height} = caps,
+        _context,
+        state
+      ) do
+    case Native.create(
+           state.text,
+           width,
+           height,
+           format,
+           state.fontsize,
+           state.box?,
+           state.boxcolor,
+           state.border?,
+           state.fontcolor,
+           fontfile_to_native_format(state.fontfile),
+           state.vertical_align,
+           state.horizontal_align
+         ) do
+      {:ok, native_state} ->
+        state = %{state | native_state: native_state}
+        {{:ok, caps: {:output, caps}, redemand: :output}, state}
+
+      {:error, reason} ->
+        raise(RuntimeError, reason)
```

Comment:
This will be `RuntimeError` anyway, so if you don't have to specify the error type, use `raise/1`
```suggestion
        raise reason
```